### ISSUE TITLE
compile failed when do  parallel build

### ISSUE THREAD

**Issue by sandy-lcq** [troglobit/sysklogd#17]
Hi,

After below  commit,  I still met one compile fail during about 2 month.
https://github.com/troglobit/sysklogd/commit/84d70e63fc105e3713943ed8c0bdd4e31a698226

This is the error log:
```
ERROR: Execution of 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/temp/run.do_compile.24857' failed with exit code 1:
make  all-recursive
make[1]: Entering directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build'
Making all in example
make[2]: Entering directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build/example'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build/example'
Making all in man
make[2]: Entering directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build/man'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build/man'
Making all in src
make[2]: Entering directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build/src'
x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o syslogd-syslogd.o `test -f 'syslogd.c' || echo '../../git/src/'`syslogd.c
x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o syslogd-socket.o `test -f 'socket.c' || echo '../../git/src/'`socket.c
x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o syslogd-timer.o `test -f 'timer.c' || echo '../../git/src/'`timer.c
../x86_64-overc-linux-libtool  --tag=CC   --mode=compile x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o libsyslog_la-syslog.lo `test -f 'syslog.c' || echo '../../git/src/'`syslog.c
x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o ../lib/pidfile.o ../../git/src/../lib/pidfile.c
x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o ../lib/strlcpy.o ../../git/src/../lib/strlcpy.c
x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o ../lib/strlcat.o ../../git/src/../lib/strlcat.c
../x86_64-overc-linux-libtool  --tag=CC   --mode=compile x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o ../lib/pidfile.lo ../../git/src/../lib/pidfile.c
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/syslog.c  -fPIC -DPIC -o .libs/libsyslog_la-syslog.o
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/../lib/pidfile.c  -fPIC -DPIC -o ../lib/.libs/pidfile.o
../x86_64-overc-linux-libtool  --tag=CC   --mode=compile x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o ../lib/strlcpy.lo ../../git/src/../lib/strlcpy.c
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/../lib/strlcpy.c  -fPIC -DPIC -o ../lib/.libs/strlcpy.o
../x86_64-overc-linux-libtool  --tag=CC   --mode=compile x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I..  -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE  -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=  -c -o ../lib/strlcat.lo ../../git/src/../lib/strlcat.c
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/../lib/strlcat.c  -fPIC -DPIC -o ../lib/.libs/strlcat.o
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/syslog.c  -fPIC -DPIC -o libsyslog_la-syslog.o >/dev/null 2>&1
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/../lib/strlcat.c  -fPIC -DPIC -o ../lib/strlcat.o >/dev/null 2>&1
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/../lib/strlcpy.c  -fPIC -DPIC -o ../lib/strlcpy.o >/dev/null 2>&1
../x86_64-overc-linux-libtool  --tag=CC   --mode=link x86_64-overc-linux-gcc  -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot=                      -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native=   -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -o syslogd syslogd-syslogd.o syslogd-socket.o syslogd-timer.o  ../lib/pidfile.o ../lib/strlcpy.o ../lib/strlcat.o 
x86_64-overc-linux-libtool: compile:  x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -DHAVE_CONFIG_H -I. -I../../git/src -I.. -DSYSCONFDIR=\"/etc\" -DRUNSTATEDIR=\"/var/run\" -D_BSD_SOURCE -D_DEFAULT_SOURCE -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -c ../../git/src/../lib/pidfile.c  -fPIC -DPIC -o ../lib/pidfile.o >/dev/null 2>&1
x86_64-overc-linux-libtool: link: x86_64-overc-linux-gcc -m64 -march=core2 -mtune=core2 -msse3 -mfpmath=sse --sysroot=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot -W -Wall -Wextra -Wno-unused-result -Wno-unused-parameter -fno-strict-aliasing -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0=/usr/src/debug/sysklogd/2.1.1-r0 -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot= -fdebug-prefix-map=TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native= -Wl,-O1 -Wl,--hash-style=gnu -Wl,--as-needed -o syslogd syslogd-syslogd.o syslogd-socket.o syslogd-timer.o ../lib/pidfile.o ../lib/strlcpy.o ../lib/strlcat.o 
TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native/usr/bin/x86_64-overc-linux/../../libexec/x86_64-overc-linux/gcc/x86_64-overc-linux/9.2.0/ld: syslogd-syslogd.o: in function `main':
/usr/src/debug/sysklogd/2.1.1-r0/build/src/../../git/src/syslogd.c:417: undefined reference to `__pidfile'
TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/recipe-sysroot-native/usr/bin/x86_64-overc-linux/../../libexec/x86_64-overc-linux/gcc/x86_64-overc-linux/9.2.0/ld: /usr/src/debug/sysklogd/2.1.1-r0/build/src/../../git/src/syslogd.c:435: undefined reference to `__pidfile'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:591: syslogd] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build/src'
make[1]: *** [Makefile:504: all-recursive] Error 1
make[1]: Leaving directory 'TOPDIR/tmp/work/core2-64-overc-linux/sysklogd/2.1.1-r0/build'
make: *** [Makefile:392: all] Error 2
WARNING: exit code 1 from a shell command.
``

**Comment by sandy-lcq** [comment]
From the log, seem there are multi process to generate pidfile.o,   so it is possible that when syslogd  try to link pidfile.o,  __pidfile is not write into it yet.

**Comment by troglobit** [comment]
That was actually quite depressing to hear. I've had an initial look at it and I'm not sure atm if it can be fixed.

**Comment by troglobit** [comment]
Fixed in PR #18

**Comment by sandy-lcq** [comment]
Hi,
I have to say that seems we still have one parallel problem,  which will not cause compile failure, but seems it is a problem,  here is the details:
dependency:
LIBOBJS: LTLIBOBJS
libsyslog.la: LTLIBOBJS

Pipeline:
|------| LTLIBOBJS
        |-------------------| libsyslog.la
	|--------| LIBOBJS
	   
Theoretically,  LIBOBJS and libsyslog.la don't have dependency, 
and their pipeline is random

Then we will face a problem, normally, LTLIBOBJS will generate
../lib/pidfile.o without fPIC, and ../lib/.lib/pidfile.o with fPIC,
but with commit https://github.com/troglobit/sysklogd/commit/c05956010c5b5517f5cbb39c2c7ad7f20ca0c1df
both ../lib/pidfile.o and ../lib/.lib/pidfile.o will compile with fPIC.

Then problem comes, LIBOBJS will generate ../lib/pidfile.o without fPIC,
and maybe cover the one with fPIC, and then be used by libsyslog.la 
(Theoretically, also random ).

Do we want libsyslog.la with fPIC both for .a and .so?
if yes, maybe we still need make LIBOBJS depend on libsyslog.la.
if not, maybe we still need make libsyslog.la depend on LIBOBJS, then it will
definately make for libsyslog.la,  .a is without fPIC and .so is with fPIC.

**Comment by troglobit** [comment]
Fixed in PR #23
